backend/tests/v5/e2e/routes/teamspaces/projects/projects.test.js

```diff
@@ -71,6 +71,155 @@ const testGetProjectList = () => {
+		});
+
+		test('should fail if the new project data are not valid', async () => {
+			const res = await agent.post(`${route}?key=${tsAdmin.apiKey}`)
+				.send({ name: 123 }).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the new project name is taken by another project', async () => {
+			const res = await agent.post(`${route}?key=${tsAdmin.apiKey}`)
+				.send({ name: testProject.name }).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should create new project if new project data are valid', async () => {
+			const res = await agent.post(`${route}?key=${tsAdmin.apiKey}`)
+				.send({ name: 'Valid Name' }).expect(templates.ok.status);
+
+			const projectsRes = await agent.get(`${route}?key=${tsAdmin.apiKey}`).expect(templates.ok.status);
+			expect(projectsRes.body.projects.find((p) => p.name === 'Valid Name')).not.toBe(undefined);
+
+			// Delete project afterwards
+			await agent.delete(`/v5/teamspaces/${teamspace}/projects/${res.body._id}?key=${tsAdmin.apiKey}`)
+				.expect(templates.ok.status);
+		});
+	});
+};
+
+const testEditProject = () => {
+	describe('Edit project', () => {
+		const route = `/v5/teamspaces/${teamspace}/projects/${testProject._id}`;
+
+		test('should fail without a valid session', async () => {
+			const res = await agent.patch(route).expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should fail without a valid teamspace', async () => {
+			const res = await agent.patch(`/v5/teamspaces/badTSName/projects/${testProject._id}?key=${tsAdmin.apiKey}`).expect(templates.teamspaceNotFound.status);
+			expect(res.body.code).toEqual(templates.teamspaceNotFound.code);
+		});
+
+		test('should fail if the user is not teamspace admin', async () => {
+			const res = await agent.patch(`${route}?key=${unlicencedUser.apiKey}`).expect(templates.teamspaceNotFound.status);
+			expect(res.body.code).toEqual(templates.teamspaceNotFound.code);
+		});
+
+		test('should fail without a valid project', async () => {
+			const res = await agent.patch(`/v5/teamspaces/teamspace/projects/12345?key=${tsAdmin.apiKey}`).expect(templates.teamspaceNotFound.status);
+			expect(res.body.code).toEqual(templates.projectNotFound.code);
+		});
+
+		test('should fail if the project data are not valid', async () => {
+			const res = await agent.patch(`${route}?key=${tsAdmin.apiKey}`)
+				.send({ name: 123 }).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the project name is taken by another project', async () => {
+			// create test project
+			const res = await agent.post(`/v5/teamspaces/${teamspace}/projects/?key=${tsAdmin.apiKey}`)
+				.send({ name: 'Existing Name' }).expect(templates.ok.status);
```

Comment:
put "Existing name" in a variable

We actually don't care what that name is so we should probably use `generateRandomString()`